src/index.js

```diff
@@ -19,70 +49,102 @@ export default class Rollup extends Plugin {
 
     this._lastBundle = null;
 
-    // TODO: maybe extract
-    this._fileToChecksumMap = Object.create(null);
-  }
-
-  writeFileIfContentChanged(fullPath, content) {
-    let previous = this._fileToChecksumMap[fullPath];
-    let next = md5Hex(content);
-
-    if (previous === next) {
-      logger.debug('cache hit, no change to: %s', fullPath);
-      // hit
-    } else {
-      logger.debug('cache miss, write to: %s', fullPath);
-      fs.writeFileSync(fullPath, content);
-      this._fileToChecksumMap[fullPath] = next; // update map
-    }
+    this._lastTree = FSTree.fromEntries([]);
   }
 
   _loadOptions() {
     // TODO: support rollup config files
     let options = this.rollupOptions;
     options.cache = this._lastBundle;
+    return options;
+  }
 
+  _targetsFor(options) {
+    if (options.dest) {
+      return [options];
+    }
     if (options.targets) {
-      return options;
-    } else {
-      options.targets = [
-        {
-          format: options.format,
-          dest: options.dest,
-          moduleName: options.moduleName
-        }
-      ];
-      delete options.format;
-      delete options.dest;
-      delete options.moduleName;
-      return options;
+      return options.targets.map(target => assign({}, options, target));
     }
+    throw new Error('missing targets or dest in options');
   }
 
-  build() {
-    const options = this._loadOptions();
-
-    if (options.entry && !path.isAbsolute(options.entry)) {
-      // if entry is not absolute, make it absolute relative to the inputPath
-      options.entry = this.inputPaths[0] + '/' + options.entry;
-    }
-
-    return heimdall.node('rollup', () => {
-      return require('rollup').rollup(options)
-        .then(bundle => {
-          this._lastBundle = bundle;
+  _patchOutput(nextTree) {
+    let patch = this._lastTree.calculatePatch(nextTree, isEqual);
+    this._lastTree = nextTree;
+    patch.forEach(([op, path, entry]) => {
+      switch (op) {
+        case 'mkdir':
+          fs.mkdirSync(this.outputPath + '/' + path);
+          break;
+        case 'rmdir':
+          fs.rmdirSync(this.outputPath + '/' + path);
+          break;
+        case 'unlink':
+          fs.unlinkSync(this.outputPath + '/' + path);
+          break;
+        case 'create':
+        case 'change':
+          fs.writeFileSync(this.outputPath + '/' + path, entry.content);
+          break;
+      }
+    });
+  }
 
-          options.targets.forEach(target => {
-            let dest = target.dest;
-            let format = target.format;
-            let output = bundle.generate(target).code;
-            let outputPath = this.outputPath + '/'+ dest;
+  _withInputPath(cb) {
+    const dir = process.cwd();
```

Comment:
Confirm, rollup uses `process.cwd()` to resolve "sources" entries in the source map.  I suppose I could rewrite them before saving the source map.  In general though, I wouldn't be surprised if rollup plugins also made this assumption.

It would be nicer if there was a cwd option that was required to be an absolute path that things were resolved against.

I realize though, if dest were resolved against the inputPath, this wouldn't matter.